### DISCUSSION TITLE
Prevent button action from running on load?

### DISCUSSION THREAD

**Discussion by gouldingken** [observablehq/framework#884] [Q&A] [answered]
I'm looking to use a button to set an input value when clicked. How do I ensure that button code is run when I click the button, but not on load? 

I tried adding `run=false` to the js header, but that prevents running altogether.

```js
const resetButton = view(Inputs.button("Reset"));
```

```js
resetButton;

inputControl.value = 222;
```

**Comment by mbostock** [comment, marked as answer]
Set the required option to true.

```js
const reset = view(Inputs.button("Reset", {required: true}));
```

This sets the initial value to undefined which will prevent the generator from resolving until the button is clicked, such that any code block referencing `reset` likewise will not until click.

**Reply by gouldingken** [reply]
Thanks - that works great. Although I realized that you also have to be careful with these code blocks rerunning based on other variables changing. But an easy workaround is to use an HTML button to ensure it only runs on click.
